python/lib/dependabot/python/update_checker/pip_compile_version_resolver.rb

```diff
@@ -143,9 +144,13 @@ def handle_pip_compile_errors(error)
           end
 
           if error.message.match?(GIT_REFERENCE_NOT_FOUND_REGEX)
-            name = error.message.match(GIT_REFERENCE_NOT_FOUND_REGEX).
-                   named_captures.fetch("name")
-            raise GitDependencyReferenceNotFound, name
+            tag = error.message.match(GIT_REFERENCE_NOT_FOUND_REGEX).named_captures.fetch("tag")
+            constraints_section = error.message.split("Finding the best candidates:").first
+            egg_regex = /#{Regexp.escape(tag)}#egg=(\S+)/
+            name_match = constraints_section.scan(egg_regex)
+            raise GitDependencyReferenceNotFound, name_match.first.first if name_match.length == 1
+
+            raise GitDependencyReferenceNotFound, tag
```

Comment:
Instead of using a bare tag name, what do you think about something like this?

```suggestion
            raise GitDependencyReferenceNotFound, "(unknown package at #{tag})"
```